pykusto/kql_converters.py

```diff
@@ -29,7 +29,7 @@ def timedelta_to_kql(td: timedelta) -> KQL:
 @kql_converter(Mapping, List, Tuple)
 def dynamic_to_kql(d: Union[Mapping, List, Tuple]) -> KQL:
```

Comment:
Do we still need this? Can't we use to_literal_dynamic?

Reply:
Yes Kusto has different behaviors for different use cases. e.g. "strcat_array" needs `to_literal_dynamic` and "in" needs `dynamic_to_kql`. 
The later is the default behavior in pykusto, and we have to manually call `to_literal_dynamic` whenever we want the other  behavior.
I think this is good enough for now, and in the future we can consider some automatic way to distinguish these cases.